src/exts/replay_handler.js

```diff
@@ -14,12 +14,15 @@ bridge.on('ReadReplayInfo', async command => {
 
 	const demo = await parser.parseDemo(sdfz);
 
-	bridge.send('ReplayInfo', {
+	const info = {
 		relativePath : command.relativePath,
 		engine: demo.header.versionString,
 		game: demo.script.gameSettings.gametype,
-		map: demo.script.gameSettings.mapname
-		// TODO: missing textual representation of start script
-		// startScript: demo.script
-	});
+		map: demo.script.gameSettings.mapname,
+		// TODO: We're passing only the alllyTeams part of the script, because
+		// that's all we need in BAR/Chobby. Do we need to pass more ?
+		allyTeams: JSON.stringify(demo.script.allyTeams),
```

Comment:
I'd rather not stringify this early...
but I've noticed that if we don't do it here, we get nonsense like [Array] or [Object]. Is there something in replay parser that's preventing proper JSON serialization?
Would an update fix it? @Jazcash

Reply:
Hi Gajop, what do you mean ? How later could you stringify ? My cursory understanding is that spring/chobby doesn't receive JSON, so if I wanted to pass something more complicated than simple key/value associations, serializing to JSON and back in chobby was the only option. Did I miss something ?

Reply:
bridge.send already converts it to string and that should work with objects/arrays of arbitrary depth.

Chobby receives it as a string and unpacks it under the assumption it's in JSON format. 

What you're doing now is stringifying it twice (once for the sub element and another for the cmd object), which I think is a bad idea. I tried to convert your code to stringify only once (i.e. let bridge.send do the serialization), but that caused some other bugs (deeply nested things came as [Array] or [Object]), which might be a bug with replay parser (perhaps it's overriding toString or toJson methods )

Reply:
Should be no problem stringifying anything from my demo parser because it's all primitive JS objects, no classes or functions in there anywhere. I've in fact been stringifying the whole of `demo` to text files since the start, for testing the output.

Reply:
@gajop after checking that the proper calls were in place in the middleware, I removed the stringify (and the `json.decode` on the lua side) and it works like a charm ... Which is really strange since it didn't work in the first place, which is why I put those calls.

Anyway, I updated the PR, now without strinfigy :)